specs/SharedBuffer.md

```diff
@@ -0,0 +1,337 @@
+This can be done by calling Close() method on the shared buffer object, or
+`chrome.webview.releaseSharedBuffer` from script.
+
+As the memory could contain sensitive information and corrupted memory could crash
+the application, the application should only share buffer with trusted sites.
+
+The application can use other messaging channel like `PostWebMessageAsJson` and
+`chrome.webview.postMessage` to inform the other side the desire to have a shared
+buffer and the status of the buffer (data is produced or consumed).
+
+# Examples
+
+The example below illustrates how to send data from application to script for one time read only consumption.
+
+The script code will look like this:
+```
+        window.onload = function () {
+            window.chrome.webview.addEventListener("SharedBufferReceived", e => {
+                SharedBufferReceived(e);
+            });
+        }
+
+        function SharedBufferReceived(e) {
+            if (e.data && e.data.read_only) {
+                // This is the one time read only buffer
+                let one_time_shared_buffer = e.sharedBuffer;
+                // Consume the data from the buffer
+                DisplaySharedBufferData(one_time_shared_buffer);
+                // Release the buffer after consuming the data.
+                chrome.webview.releaseSharedBuffer(one_time_shared_buffer);
+            }
+        }
+```
+## Win32 C++
+```cpp
+
+        wil::com_ptr<ICoreWebView2Environment11> environment;
+        CHECK_FAILURE(
+            m_appWindow->GetWebViewEnvironment()->QueryInterface(IID_PPV_ARGS(&environment)));
+
+        wil::com_ptr<ICoreWebView2SharedBuffer> sharedBuffer;
+        CHECK_FAILURE(environment->CreateSharedBuffer(bufferSize, &sharedBuffer));
+        // Fill data into the shared memory via IStream.
+        wil::com_ptr<IStream> stream;
+        CHECK_FAILURE(sharedBuffer->GetStream(&stream));
+        CHECK_FAILURE(stream->Write(data, dataSize, nullptr));
+        PCWSTR additionalDataAsJson = L"{\"read_only\":true}";
+        if (forFrame)
+        {
+            m_webviewFrame->PostSharedBufferToScript(
+                sharedBuffer.get(), /*isReadOnlyToScript*/TRUE, additionalDataAsJson);
+        }
+        else
+        {
+            m_webView->PostSharedBufferToScript(
+               sharedBuffer.get(), /*isReadOnlyToScript*/TRUE, additionalDataAsJson);
+        }
+        // Explicitly close the one time shared buffer to ensure that the resource is released timely.
+        sharedBuffer->Close();
+        
+```
+## WinRT and .NET
+```c#
+      using (CoreWebView2SharedBuffer sharedBuffer = WebViewEnvironment.CreateSharedBuffer(bufferSize))
+      {
+          // Fill data using access Stream
+          using (Stream stream = sharedBuffer.GetStream())
+          {
+              using (StreamWriter writer = new StreamWriter(stream))
+              {
+                  writer.Write(bufferData);
```

Comment:
What is bufferData? Should this be sharedBuffer?

Reply:
it is actually the data to write into the buffer. I'll rename it as data, to be the same as C++ sample.